### PR TITLE
Iowa Itemized and Standard Deduction Changes for 2023

### LINKED ISSUE #4935: Iowa Itemized and Standard Deduction Changes for 2023
Iowa repealed its own standard deduction and now incorporates the federal standard deduction (Ex. single filer amount went from $2,210 to $13,850).

[Reference](https://revenue.iowa.gov/taxes/tax-guidance/individual-income-tax/1040-expanded-instructions/whats-new)

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #4935

### CODE REVIEW

File: policyengine_us/variables/gov/states/ia/tax/income/deductions/ia_itemized_deductions_unit.py [hunk 1]
```diff
@@ -24,6 +24,18 @@ def formula(tax_unit, period, parameters):
         federal dollar limitation.
         """
         # compute tax unit's itemized deductions
-        itm_deds_less_salt = tax_unit("itemized_deductions_less_salt", period)
-        uncapped_property_taxes = add(tax_unit, period, ["real_estate_taxes"])
-        return itm_deds_less_salt + uncapped_property_taxes
+        p = parameters(period).gov.states.ia.tax.income.deductions.itemized
+        if p.fed_ded_applies:
+            p_fed = parameters(period).gov.irs.deductions
+            itemized_deductions = add(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ia/tax/income/deductions/ia_itemized_deductions_unit.py:30]:
```suggestion
            return add(
```

File: policyengine_us/variables/gov/states/ia/tax/income/deductions/ia_itemized_deductions_unit.py [hunk 2]
```diff
@@ -24,6 +24,18 @@ def formula(tax_unit, period, parameters):
...
+                tax_unit, period, p_fed.itemized_deductions
+            )
+        else:
+            itm_deds_less_salt = tax_unit(
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ia/tax/income/deductions/ia_itemized_deductions_unit.py:34]:
```suggestion
            return add(tax_unit, period, ["itemized_deductions_less_salt", "real_estate_taxes"])
```

File: policyengine_us/tests/policy/baseline/gov/states/ia/tax/income/ia_withheld_income_tax.yaml [hunk 3]
```diff
@@ -27,3 +27,13 @@
     state_code: IA
   output:
     ia_withheld_income_tax: 0
+
+- name: Federal feduction post 2023
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/states/ia/tax/income/ia_withheld_income_tax.yaml:31]:
```suggestion
- name: Federal deduction from 2023 on
```

File: policyengine_us/parameters/gov/states/ia/tax/income/deductions/itemized/fed_ded_applies.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+values:
+  0000-01-01: false
+  2023-01-01: true
+
+metadata:
+  label: Iowa federal itemized deduction applies
+  period: year
+  unit: bool
+  reference: 
+  - title: Iowa Department of Revenue - 2023 IA1040 Income Tax Return instructions - Iowa Taxable Income
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ia/tax/income/deductions/itemized/fed_ded_applies.yaml:11]: Legal code?

File: policyengine_us/parameters/gov/states/ia/tax/income/deductions/itemized/fed_ded_applies.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ia/tax/income/deductions/itemized/fed_ded_applies.yaml:1]: rename `applies_federal` (i think we don't usually shrink words like ded in files right)

File: changelog_entry.yaml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Apply the federal standard and itemized deductions in Iowa post 2023.
```
Comment (MaxGhenis) [inline changelog_entry.yaml:4]:
```suggestion
    - Apply the federal standard and itemized deductions in Iowa from 2023 on.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/ia/tax/income/deductions/ia_standard_deduction_indiv.py [hunk 6]
```diff
@@ -36,5 +36,12 @@ def formula(person, period, parameters):
...
-        return (is_head | is_spouse) * p.deductions.standard[filing_status]
+        p = parameters(period).gov.states.ia.tax.income.deductions.standard
+
+        if p.applies_federal:
+            fed_p = parameters(period).gov.irs.deductions
+            deduction = fed_p.standard.amount[filing_status]
+        else:
+            deduction = p.amount[filing_status]
+
+        return (is_head | is_spouse) * deduction
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/ia/tax/income/deductions/ia_standard_deduction_indiv.py:47]:
```suggestion
        return tax_unit("is_tax_unit_head_or_spouse", period) * deduction
```

File: policyengine_us/parameters/gov/states/ia/tax/income/deductions/itemized/applies_federal.yaml [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+  2023-01-01: true
+
+metadata:
+  label: Iowa federal itemized deduction applies
+  period: year
+  unit: bool
+  reference: 
+  - title: Iowa Department of Revenue - 2023 IA1040 Income Tax Return instructions - Iowa Taxable Income
+    href: https://revenue.iowa.gov/taxes/tax-guidance/individual-income-tax/1040-expanded-instructions/iowa-taxable-income
+  - title: Iowa Individual Income, Corporate, and Franchise Taxes, §422.5 3(a)
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/ia/tax/income/deductions/itemized/applies_federal.yaml:13]: I don't see this here. It talks about adding federal deductions back for the purposes of, I think, some state tax deduction? Can you explain in a comment if it's here?

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/ia/tax/income/deductions/itemized/applies_federal.yaml (hunk 7 above)
Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/ia/tax/income/deductions/itemized/applies_federal.yaml]: You are right, 

I found the legal code but it does not seem to [represent the changes](https://casetext.com/regulation/iowa-administrative-code/agency-701-revenue-department/title-iv-individual-income-tax/chapter-303-determination-of-taxable-income/rule-701-3034-optional-standard-deduction) or at least explicitly state the change in 2023 - have added a comment - the instructions are clear that the federal ded should be used

Comment (MaxGhenis) [reply policyengine_us/parameters/gov/states/ia/tax/income/deductions/itemized/applies_federal.yaml]: where does the law define the standard deduction values pre-2023?

**Review by MaxGhenis — APPROVED** [review]

**Comment by MaxGhenis** [comment]
please git pull upstream master to get the tests running again

**Pull request merged**